### ISSUE TITLE
Concurrency seems to break non-concurrent behaviour 

### ISSUE THREAD

**Issue by tgermain** [NicolasLM/sauna#53]
I tried to run sauna with the latest docker image (and the latest commit from master) and I can't get it to start with `concurrency: 1`

# Conf used : 
```yaml
---                                                                
periodicity: 10                                                    
concurrency: 1                                                     
hostname: dev-timothee                                             
                                                                   
consumers:                                                         
  # Listen on a TCP port and serve results to incoming connections 
  - type: TCPServer                                                
    port: 55555                                                    
  # Just prints checks on the standard output                      
  - type: Stdout                                                   
  # HTTP Server that exposes sauna status as a REST API
  - type: HTTPServer
    port: 8088

plugins:
  # Fake checks that return the provided
  # status and output
  - type: Dummy
    checks:
      - type: dummy
        status: 0
        output: Everything is alright

```
# result : 
```
$  sauna --level debug
2017/01/09 15:55:42 - INFO     - Running checks with interval: {10}
2017/01/09 15:55:42 - DEBUG    - Running consumer TCPServer
2017/01/09 15:55:42 - DEBUG    - Scheduler has 1 ticks, each one is 10 seconds
2017/01/09 15:55:42 - DEBUG    - Tick 0, scheduled [<Job launch_and_publish_checks_with_periodicity every 10 seconds>]
2017/01/09 15:55:42 - DEBUG    - Pushing to consumers: dummy_dummy
Exception in thread producer:
Traceback (most recent call last):
  File "/home/admin/.pyenv/versions/3.5.2/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/home/admin/.pyenv/versions/3.5.2/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/home/admin/dev/sauna/sauna/__init__.py", line 331, in run_producer
    for _ in scheduler:
  File "/home/admin/dev/sauna/sauna/scheduler.py", line 54, in __next__
    job()
  File "/home/admin/dev/sauna/sauna/scheduler.py", line 97, in __call__
    self.func(*self.func_args, **self.func_kwargs)
  File "/home/admin/dev/sauna/sauna/__init__.py", line 291, in launch_and_publish_checks_with_periodicity
    self._check_helper(check)
  File "/home/admin/dev/sauna/sauna/__init__.py", line 302, in _check_helper
    self._current_checks.remove(service_check.name)
ValueError: list.remove(x): x not in list

2017/01/09 15:55:42 - DEBUG    - [StdoutConsumer] Got check ServiceCheck(timestamp=1483977342, hostname='dev-timothee', name='dummy_dummy', status=0, output='Everything is alright')
ServiceCheck(timestamp=1483977342, hostname='dev-timothee', name='dummy_dummy', status=0, output='Everything is alright')
2017/01/09 15:55:42 - DEBUG    - Running consumer Stdout
2017/01/09 15:55:42 - DEBUG    - Running consumer HTTPServer
2017/01/09 15:55:42 - INFO     - Exiting...
2017/01/09 15:55:42 - INFO     - [StdoutConsumer] Check sent
2017/01/09 15:55:42 - DEBUG    - [StdoutConsumer] Exited consumer thread
2017/01/09 15:55:43 - DEBUG    - [HTTPServerConsumer] Exited consumer thread
2017/01/09 15:55:43 - DEBUG    - [TCPServerConsumer] Exited consumer thread
2017/01/09 15:55:43 - DEBUG    - Exited main thread
```

It's seems to be related to this line : 
https://github.com/NicolasLM/sauna/blob/master/sauna/__init__.py#L300

# What I try to make it work : 
Check if the service is in the _current_check list before trying to remove it
```python
with self._current_checks_lock:                    
    if service_check.name in self._current_checks:
        self._current_checks.remove(service_check.name)
```

It seems to work

**Comment by NicolasLM** [comment]
Hi Timothée,

I pushed a fix for this issue in #54. If that's good for you, Loïc can merge it.

**Comment by tgermain** [comment]
It works, you can go on and merge @bewiwi
